CapstoneProject/Assets/Scripts/Core/BehaviorTree/AI.cs

```diff
@@ -151,6 +175,52 @@ public bool IsWithinRange()
         return result;
     }
 
+    [Task]
+    public bool IsWithinRange(float range)
+    {
+        bool result = false;
+        float distanceToTarget = Vector2.Distance(playerPos.position, transform.position);
+        float attackRange = range;
+
+        if (distanceToTarget <= attackRange)
+        {
+            if (playerPos.position.x < transform.position.x)
+            {
+                if (_rb.CurrentFacingDirection == RigidbodyComponent.FacingDirections.RIGHT)
+                {
+                    Turn();
+                }
+            }
+            else
+            {
+                if (_rb.CurrentFacingDirection == RigidbodyComponent.FacingDirections.LEFT)
+                {
+                    Turn();
+                }
+
+            }
+            result = true;
+        }
+        return result;
+    }
+
+    [Task]
+    public void Chase()
+    {
+        movementSpeed = initialSpeed;
+        AnimStateUpdate();
+        transform.position = Vector2.MoveTowards(transform.position, playerPos.position, movementSpeed * Time.deltaTime);
+        Task.current.Succeed();
+    }
+    [Task]
+    public void ReturnToSpawnPoint()
+    {
+        movementSpeed = initialSpeed;
+        AnimStateUpdate();
+        transform.position = Vector2.MoveTowards(transform.position, SpawnPoint, movementSpeed * Time.deltaTime);
+        Task.current.Succeed();
+    }
```

Comment:
these both do the same things with only the position to go to changing, I'd make a sub function called MoveTo and have a param for the position and then call in each of these.